### ISSUE TITLE
extrude svg to curb geometry

### ISSUE THREAD

**Issue by kfarr** [3DStreet/3dstreet#473]
As a user creating a custom scene with curves or other custom roadway geometry, I need to import an svg path from a third-party editor such as Figma, copy/paste the SVG code into a 3dstreet component, the component should then extrude the svg and apply repeating texture in the same style as linear 3DStreet segments such as sidewalk, street asphalt or concrete, and grass.

helpful docs:
https://threejs.org/docs/index.html#api/en/geometries/ExtrudeGeometry

Related stack overflow question about error encountered while attempted to extrude svg into mesh:
https://stackoverflow.com/questions/69879688/three-js-extrudegeometry-issue-for-svg-containing-ring

Example figma scene from which svg will be exported
https://www.figma.com/file/I0TER1qcjoRKu2m4zJdaCE/Mission-Bay-school-traffic-circle?type=design&node-id=7-11&mode=design&t=YpJiCWU8Lq7gyDOU-0

Example of exported SVG file from this figma scene to be supported by this issue. The filled white polygon paths should be extruded as sidewalk or grass "segment type"
![Traffic Circle Vector for SVG Export](https://github.com/3DStreet/3dstreet/assets/470477/d87ef92d-af6a-46a8-825e-34c05f13b9db)

Actual working example with this svg:
https://codepen.io/kfarr/pen/LYaadRG (based on this [original codepen](https://codepen.io/areknawo/pen/abWzjGO))

@Algorush Suggested order of operations:
- [ ] adapt the kfarr codepen above to an a-frame component that works with latest version of a-frame / three.js (this might "just work" or it might need more effort)
- [ ] add component property for svg string with no default value (empty / no mesh rendered by default)
- [ ] create example 3dstreet scene with this component and the contents of the svg above and 1 streetmix street to prove it can coexist in one scene
- [ ] update material options to match user story at top of this issue
- [ ] retain the codepen blue mesh with blue outline as a "blueprint" or similar material option (or variant type) for the future

**Comment by Algorush** [comment]
Testing in console with these commands:
```
tc = document.createElement('a-entity')
tc.setAttribute('svg-extruder', `svgString: <svg id="traffic-circle-svg" width="1562" height="1722" viewBox="0 0 1562 1722" fill="none" xmlns="http://www.w3.org/2000/svg">
            <circle cx="747" cy="884" r="376" fill="white"/>
            <path d="M170 921C110.447 960.339 73.1658 978.46 0 1004L24 1070.5C87.6715 1053.2 126.721 1054.53 200.5 1070.5C180.691 1013.73 173.793 981.04 170 921Z" fill="white"/>
            <path d="M920.5 16.5L873 0C826.761 111.5 798.791 179.933 747 307.5C818.049 307.14 904.5 334 904.5 334C896.322 273.342 871.658 119.714 920.5 16.5Z" fill="white"/>
            <path d="M1562 797C1475.23 805.17 1419.94 800.652 1310 777C1322.14 822.934 1324.73 853.264 1326 911C1426.16 863.684 1479.82 844.12 1562 847V797Z" fill="white"/>
            <path d="M832 1467C782.879 1472.52 753.742 1472.69 697 1467C729.414 1550.35 751.819 1619.31 761 1722H803.5C806.545 1646.07 790.668 1543.99 832 1467Z" fill="white"/>
        </svg>`)
tc.setAttribute('data-layer-name', 'Segment • SVG-Extrude')
tc.setAttribute('scale', '0.05 0.05 0.05')
sp = document.querySelector('.street-parent')
sp.appendChild(tc)
```
Here also need to scale it down to 0.05 to fit the scene size

**Comment by kfarr** [comment]
wow it's working great so far! 

![image](https://github.com/3DStreet/3dstreet/assets/470477/518be2ac-3087-43dc-b0b2-88a0fe889ec5)

**Comment by Algorush** [comment]
now extruded svg works with applied texture
![image](https://github.com/3DStreet/3dstreet/assets/32197767/42312cc8-aafe-4cf1-a604-63e3e752bd7e)

**Comment by Algorush** [comment]
I think it also will be good to add elevation level property in this component. So that the user can quickly set the height (y-position) to match existing streetmix and 3DStreet levels

**Comment by kfarr** [comment]
closed by #478
